test/concerns/sti_support_test.rb

```diff
@@ -35,4 +35,26 @@ def test_sti_support_with_from_subclass
       Object.send :remove_const, 'SubclassWithAncestry'
     end
   end
+
+  def test_sti_support_for_counter_cache
+    AncestryTestDatabase.with_model :counter_cache => true, :extra_columns => {:type => :string} do |model|
+      Object.const_set 'Subclass3', Class.new(model)
+      Object.const_set 'Subclass4', Class.new(model)
+
+      node1 = Subclass3.create!
+      node2 = Subclass4.create! :parent => node1
+      node3 = Subclass3.create! :parent => node1
+      node4 = Subclass4.create! :parent => node3
+      node5 = Subclass3.create! :parent => node4
+
+      assert_equal 2, node1.reload.children_count
+      assert_equal nil, node2.reload.children_count
```

Comment:
odd that this would be `nil` and not 0.
Do we need to put in a `to_i` in there or something?

Do you have a theory why it is happening?

Reply:
I _think_ it's because we didn't set a default value when setting up the children_count column on the `test_nodes` table https://github.com/stefankroes/ancestry/blob/1d0255133777f6710af32ae7b89a85aa62fa3971/test/environment.rb#L77